src/dmqmc_initialise_dm.F90

```diff
@@ -463,7 +467,11 @@ subroutine initialise_dm_metropolis(sys, rng, qmc_state, dmqmc_in, npsips, spawn
                         call gen_excit_ptr%full(rng, sys, qmc_state%excit_gen_data, cdet, pgen, connection, hmatel, allowed)
                         ! Check that we didn't generate a null excitation.
                         ! [todo] - Modify accordingly if pgen is ever calculated in for the ueg.
-                        if (abs(hmatel%r) < depsilon) cycle
+                        if (sys%read_in%comp) then
+                            if (abs(hmatel%c) < depsilon) cycle
+                        else
+                            if (abs(hmatel%r) < depsilon) cycle
+                        end if
```

Comment:
Is this the correct check for a null excitation for complex excitation generation?

Reply:
I'm asking myself this btw

Reply:
So a null excitation will set both hmatel%r and hmatel%c to zero. Currently you're checking if either the real part or the imaginary part are zero.

Reply:
In `hamiltonian_data.f90`, it says `hmatel%c` is value for complex Hamiltonians instead of the imaginary part, and routines in `hamiltonian_periodic_complex.f90` set only `hmatel%c` instead of both. Maybe the zero result of `hmatel%r` is caused by variable initialization?

Reply:
so hmatel%c is a complex number.

Reply:
Often in excitation generators, if an excitation is not valid, we set both the real and the complex hmatel to zero because we don't know what calculation called the excitation generator.

Reply:
Yes, `hmatel%c` is of type `complex(p)`.

But I see the complex Hamiltonian evaluation routines set only `hmatel%c` hence it seems to be safer to check only `hmatel%c` to avoid non-zero variable initialization issues.

Reply:
Ah, sorry, I misunderstood what hmatel%r and hmatel%c meant.